jme3-core/src/main/java/com/jme3/anim/tween/action/Action.java

```diff
@@ -33,38 +127,62 @@ public double getLength() {
     /**
      * Alter the length (duration) of this Action.  This can be used to extend
      * or truncate an Action.
-     *
      * @param length the desired length (in unscaled seconds, default=0)
      */
     public void setLength(double length) {
         this.length = length;
     }
 
+    /**
+     * Gets the speed of the frames of this action which gets applied for the Layer enclosing this action.
+     * Gets applied each advance() by this formula : time += tpf * currentAction.getSpeed() * ac.globalSpeed.
+     * @return the speed of frames.
+     */
     public double getSpeed() {
         return speed;
     }
 
+    /**
+     * Sets the speed of the frames of this action which gets applied for the Layer enclosing this action.
+     * Gets applied each advance() by this formula : time += tpf * currentAction.getSpeed() * ac.globalSpeed.
+     * @param speed the speed of frames.
+     */
     public void setSpeed(double speed) {
         this.speed = speed;
-        if( speed < 0){
-            setForward(false);
-        } else {
-            setForward(true);
-        }
+        setForward(speed < 0);
     }
 
+    /**
+     * Gets the animation mask for this action.
+     * the animation mask holds data(subsets of an object) about which part of the animation would be played.
+     * @return the animation mask instance.
+     */
     public AnimationMask getMask() {
         return mask;
     }
 
+    /**
+     * Sets the animation mask for this action.
+     * the animation mask holds the info about which part(subsets of an object) of the animation to play, like joints in #{@link com.jme3.anim.ArmatureMask#addFromJoint(Armature, String)}
```

Comment:
Add a space between "data" and "("
Change "like" to "such as",  which is more precise language.